frappe/public/js/frappe/form/multi_select_dialog.js

```diff
@@ -1,110 +1,76 @@
-// Copyright (c) 2015, Frappe Technologies Pvt. Ltd. and Contributors
-// MIT License. See license.txt
-
 frappe.ui.form.MultiSelectDialog = Class.extend({
-	init: function(opts) {
-		/* Options: doctype, target, setters, get_query, action */
+	init: function (opts) {
+		/* Options: doctype, target, setters, get_query, action, add_filters_group, data_fields, primary_action_label */
 		$.extend(this, opts);
-
 		var me = this;
-		if(this.doctype!="[Select]") {
-			frappe.model.with_doctype(this.doctype, function(r) {
+		if (this.doctype != "[Select]") {
+			frappe.model.with_doctype(this.doctype, function () {
 				me.make();
 			});
 		} else {
 			this.make();
 		}
 	},
-	make: function() {
+	make: function () {
 		let me = this;
 
 		this.page_length = 20;
 		this.start = 0;
 
-		let fields = [
-			{
-				fieldtype: "Data",
-				label: __("Search Term"),
-				fieldname: "search_term"
-			},
-			{
-				fieldtype: "Column Break"
-			}
-		];
-		let count = 0;
-		if(!this.date_field) {
-			this.date_field = "transaction_date";
-		}
+		let fields = [];
 
-		// setters can be defined as a dict or a list of fields
-		// setters define the additional filters that get applied
-		// for selection
+		let header_fields = this.make_dialog_header();
 
-		// CASE 1: DocType name and fieldname is the same, example "customer" and "customer"
-		// setters define the filters applied in the modal
-		// if the fieldnames and doctypes are consistently named,
-		// pass a dict with the setter key and value, for example
-		// {customer: [customer_name]}
-
-		// CASE 2: if the fieldname of the target is different,
-		// then pass a list of fields with appropriate fieldname
+		fields = fields.concat(header_fields);
```

Comment:
`fields` is already declared as an empty array and added here. Instead just assign the value to fields directly.

```suggestion
		let fields = this.make_dialog_header();
```

Reply:
done